libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -77,6 +80,8 @@ void DataFlowAnalyzer::operator()(ExpressionStatement& _statement)
 		cxx20::erase_if(m_memory, mapTuple([&](auto&& key, auto&& /* value */) {
 			return !m_knowledgeBase.knownToBeDifferentByAtLeast32(vars->first, key);
 		}));
+		// TODO erase keccak knowledge, but in a more clever way
+		m_keccak = {};
```

Comment:
Ouch, so writing anything anywhere in memory just discards all the info about keccak? I'd expect it to limit the usefulness of this step a lot. Basically it will not help much beyond just situations where there are keccak calls just next to each other?

Reply:
Yes, that's why there is the TODO.